src/writer/ai.py

```diff
@@ -1824,7 +1824,18 @@ def _is_serialized(self, message: 'Conversation.Message') -> bool:
         elif message.get("tool_call_id") is not None:
             return False
         tool_calls = message.get("tool_calls")
-        if tool_calls is not None and tool_calls != []:
+        if (
```

Comment:
Is there a reason to format like that instead of just doing what follow ?

```python
if tool_calls is not None and tool_calls != [] and not message.get("content"):
    # If tool call request message
    # doesn't have meaningful content,
    # we don't serialize it
    return False
```

Reply:
I wanted to separate those checks semantically, to indicate that they are the two components for a successful validation - "tool calls are present" and "no content is available". But you're right to point this out, now that I look at it, it looks redundant. Probably this should be even `if not tool_calls and not message.get('content')`.

Reply:
My two cents: I think it's more readable to introduce a meaningful variable which document the code itself. It might even make the comments not needed.
```py
has_tool_calls = tool_calls is not None and tool_calls != []

if has_tool_calls and not message.get("content"):
    return False
```